cgroups/systemd/apply_systemd.go

```diff
@@ -212,6 +213,17 @@ func (m *Manager) Apply(pid int) error {
 
 	m.Paths = paths
 
+	cpuShares, err := ioutil.ReadFile(path.Join(m.Paths["cpu"], "cpu.shares"))
+	if err != nil {
+		return err
+	}
+
+	if c.CpuShares != 0 {
+		if strings.TrimSpace(strconv.FormatInt(c.CpuShares, 10)) != strings.TrimSpace(string(cpuShares)) {
```

Comment:
yeah I used TrimSpace because even though the values {1: Set by the kernel, 2: Passed by docker run} were same ... the leading and trailing spaces were failing the comparisons.